### ISSUE TITLE
EXC_BAD_ACCESS crash when push notification clicked

### ISSUE THREAD

**Issue by sercanov** [WebEngage/react-native-webengage#3]
We are using both WebEngage and Firebase in our app. 
iOS apps are crashing when Firebase push notification clicked. 
Here is stack trace of the crash logs;

EXC_BAD_ACCESS-[WEGPushNotificationHandler handlePushNotification:withCtaId:]

<img width="1082" alt="screen shot 2018-10-05 at 11 08 49" src="https://user-images.githubusercontent.com/10291644/46523650-17349280-c88f-11e8-94bd-8df48795a1ab.png">

**Comment by sercanov** [comment]
Here is my AppDelegate related sections.
I am also migrating it to example app initialization of example app in this repository. Don't know if its related

```
 (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  
  [FIRApp configure];
  [RNFirebaseNotifications configure];
  
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  self.window.backgroundColor = [UIColor whiteColor];
  
  [[WebEngage sharedInstance] application:application didFinishLaunchingWithOptions:launchOptions autoRegister:NO];

		NSURL *jsCodeLocation;
		#ifdef DEBUG
			jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
		#else
			jsCodeLocation = [CodePush bundleURL];
		#endif
		
		[[RCCManager sharedInstance] initBridgeWithBundleURL:jsCodeLocation launchOptions:launchOptions];

		return YES;
}
  

- (void)applicationDidBecomeActive:(UIApplication *)application {
  [FBSDKAppEvents activateApp];
}

- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
  
}

- (void)application:(UIApplication *)application didReceiveLocalNotification:(UILocalNotification *)notification {
  [[RNFirebaseNotifications instance] didReceiveLocalNotification:notification];
}

- (void)application:(UIApplication *)application didReceiveRemoteNotification:(nonnull NSDictionary *)userInfo
                                                       fetchCompletionHandler:(nonnull void (^)(UIBackgroundFetchResult))completionHandler{
  [[RNFirebaseNotifications instance] didReceiveRemoteNotification:userInfo fetchCompletionHandler:completionHandler];
}

- (void)application:(UIApplication *)application didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings {
  [[RNFirebaseMessaging instance] didRegisterUserNotificationSettings:notificationSettings];
}


- (BOOL)application:(UIApplication *)application
            openURL:(NSURL *)url
  sourceApplication:(NSString *)sourceApplication
         annotation:(id)annotation {
  return [[FBSDKApplicationDelegate sharedInstance] application:application
                                                        openURL:url
                                              sourceApplication:sourceApplication
                                                     annotation:annotation];
}

`

**Comment by saumitrabhave** [comment]
Hi @sercanov , Can you help me with following 
1. your podfile.lock line for WebEngage
2. Initialization code for your app `https://developer.apple.com/documentation/usernotifications/unusernotificationcenterdelegate`, I mean where is it initialized and set to `UNUserNotificationCenter.delegate` property.

**Comment by sercanov** [comment]
Hi @saumitrabhave 

Podfile.lock
```
  - WebEngage (4.0.19):
    - WebEngage/Xcode9 (= 4.0.19)
  - WebEngage/Xcode9 (4.0.19)
```

2. Actually we didn't initialized anything like `UNUserNotificationCenter.delegate` 
We followed react-native-firebase instructions for it, you can see it here https://rnfirebase.io/docs/v5.x.x/notifications/ios

**Comment by saumitrabhave** [comment]
Hi @sercanov , How is the order of Initialization for `[FIRApp configure]` and `[RNFirebaseNotifications configure];` and `[[WebEngage sharedInstance] application:application didFinishLaunchingWithOptions:launchOptions autoRegister:NO];`

can you try switching the Initialization orders and see if that fixes the issue for you.

**Comment by sercanov** [comment]
hey @saumitrabhave 

After I migrated to ExampleApp, now I am getting crash on WebEngage push. No more EXC_BAD_ACCESS I suppose.

New crash is;

*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSDictionaryM setObject:forKeyedSubscript:]: key cannot be nil'
***


Here is whole AppDelegate, I tried moving [WebEngage sharedInstance] to the just above return YES. Crash is the same

```obj-c
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  
  self.bridge = [WEGWebEngageBridge new];
  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self.bridge launchOptions:launchOptions];
  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge moduleName:@"Scotty" initialProperties:nil];
  
  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];
  
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];
  
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  self.window.backgroundColor = [UIColor whiteColor];
  
  [GMSServices provideAPIKey:@"XXX"];
  [FIRApp configure];
  [RNFirebaseNotifications configure];
  [[WebEngage sharedInstance] application:application didFinishLaunchingWithOptions:launchOptions autoRegister:NO];
  
  [ChatStyling applyStyling];
  [ZDCChat initializeWithAccountKey:@"XXX"];
  [[[ZDCChat instance] overlay] setEnabled:NO];
  
  [RNSentry installWithBridge:[[RCCManager sharedInstance] getBridge]];
  
  NSURL *jsCodeLocation;
#ifdef DEBUG
  jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
#else
  jsCodeLocation = [CodePush bundleURL];
#endif
  
  [[RCCManager sharedInstance] initBridgeWithBundleURL:jsCodeLocation launchOptions:launchOptions];
  
  return YES;
}
```

**Comment by saumitrabhave** [comment]
@sercanov , Glad the EXEC_BAD_ACCESS is resolved, can you please share the new stacktrace.

**Comment by ygit** [comment]
@sercanov If this issue still persists, can you update to the iOS SDK version **4.5.0**?
Let us know your feedback.

**Comment by ygit** [comment]
The reported issue has been resolved in iOS SDK version **4.5.0**
